binding.cc

```
@@ -952,28 +995,46 @@ namespace zmq {
     if(zmq_socket_monitor(socket->socket_, addr, ZMQ_EVENT_ALL) != -1) {
       socket->monitor_socket_ = zmq_socket (context->context_, ZMQ_PAIR);
       zmq_connect (socket->monitor_socket_, addr);
-
+      socket->timer_interval_ = timer_interval;
+      socket->num_of_events_ = num_of_events;
       socket->monitor_handle_ = new uv_timer_t;
-
       socket->monitor_handle_->data = socket;
 
       uv_timer_init(uv_default_loop(), socket->monitor_handle_);
-      uv_timer_start(socket->monitor_handle_, reinterpret_cast<uv_timer_cb>(Socket::UV_MonitorCallback), timer_interval, timer_interval);
+      uv_timer_start(socket->monitor_handle_, reinterpret_cast<uv_timer_cb>(Socket::UV_MonitorCallback), timer_interval, 0);
     }
 
     NanReturnUndefined();
   }
 
+  void
+  Socket::Unmonitor() {
+    // Make sure we are monitoring
+    if (this->monitor_socket_ == NULL) {
+      return;
+    }
+
+    // Passing NULL as addr will tell zmq to stop monitor
+    zmq_socket_monitor(this->socket_, NULL, ZMQ_EVENT_ALL);
+
+    // Close the monitor socket and stop timer
+    if (zmq_close(this->monitor_socket_) < 0)
+      throw std::runtime_error(ErrorMessage());
```

I didn't touch that, whether or not we want this module to throw errors is a good question, I think that we should just raise an error event and let the use close the socket and create a new one. Throwing exception could crash the user process, which is a little too much...

My concern is L1018, calling `zmq_socket_monitor` with NULL address in order to stop monitor, I didn't find anything in the docs, only in the zmq code: https://github.com/zeromq/libzmq/blob/master/src/socket_base.cpp#L1380